cocos/animation/marionette/pose-graph/foundation/authoring/input-authoring.ts

```diff
@@ -0,0 +1,453 @@
+    displayName?: string;
+}
+
+interface ArrayPropertySyncGroup {
+    members: string[];
+}
+
+export interface PoseGraphNodeInputMetadata {
+    type: PoseGraphType;
+
+    displayName?: string;
+
+    deletable?: boolean;
+
+    insertPoint?: boolean;
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type Constructor = Function;
+
+interface PropertyNodeInputRecord {
+    type: PoseGraphType;
+    displayName?: string;
+    getArrayElementDisplayName?(index: number): string;
+    arraySyncGroup?: ArrayPropertySyncGroup;
+}
+
+export type PoseGraphNodeInputInsertId = string;
+
+/**
+ * @zh 描述一个姿势图结点类属性在映射为结点输入时的选项。
+ * @en Describes the options used
+ * when a pose node class property is going to be mapped as node input(s).
+ */
+export interface NodeInputMappingOptions {
+    /**
+     * @zh 此属性或属性的元素映射的输入类型。
+     * 输入的类型必须和属性（的元素）的实际值类型匹配。
+     *
+     * @en Type of the input(s) that the property associates.
+     * The input type should be matched with the actual value type of the property.
+     */
+    type: PoseGraphType;
+
+    /**
+     * @zh 此属性关联的输入的显示名称。
+     * @en Display of the input(s) that the property associates.
+     */
+    displayName?: string;
+
+    /**
+     * @zh 若有定义，当该属性是数组时，此方法被用来获取数组各个元素所映射的输入的显示名称。
+     * @en If specified and if the property is an array,
+     * this method will be used to retrieve the display name of input mapped by each element.
+     * @param index 数组元素的索引。
+     */
+    getArrayElementDisplayName?(index: number): string;
+
+    /**
+     * @zh 若有定义，当该属性是数组时，声明该数组属性属于指定的同步组。
+     * 同步组中的任何成员在进行输入增加、移除或挪动等操作时，其它成员也将进行该操作。
+     *
+     * @en If specified and if the property is an array,
+     * declares that the array property belongs to specified sync group.
+     * When any member of the sync group is performing input add/remove/move etc. operation,
+     * the other members will also perform that operation.
+     *
+     * @example
+     * @zh
+     * 对于如下结点类来说：
+     *
+     * ```ts
+     * class Blending extends PoseNode {
+     *     \@input({ type: PoseGraphType.POSE, syncGroup: 'blend-item' })
+     *     poses: PoseNode[] = [];
+     *
+     *     \@input({ type: PoseGraphType.FLOAT, syncGroup: 'blend-item' })
+     *     weights: number[] = [];
+     * }
+     * ```
+     *
+     * 当 `poses` 插入了一个新的输入后，`weights` 也会被插入一个新输入。
+     *
+     * @en
+     * For the following node class:
+     *
+     * ```ts
+     * class Blending extends PoseNode {
+     *     \@input({ type: PoseGraphType.POSE, syncGroup: 'blend-item' })
+     *     poses: PoseNode[] = [];
+     *
+     *     \@input({ type: PoseGraphType.FLOAT, syncGroup: 'blend-item' })
+     *     weights: number[] = [];
+     * }
+     * ```
+     *
+     * If a new input is inserted into `poses`, there will be also a new input inserted into weights.
+     */
+    arraySyncGroup?: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+class NodeInputManager {
```

Comment:
PoseGraphNodeInputManager seems more clear.

Reply:
Should we prefix all node related a `PoseGraphNode`? For example exported methods in https://github.com/cocos/cocos-engine/pull/14952/files/4d734418971df040a536acf0e30698b749e76d8e#diff-fef71171b3da4eb1de158627c4eaca12a9f456a43417c06075f2bb27814ad755 .

Reply:
Since I noticed that some other types have added prefixes, but this NodeInputManager has no prefix, so it's better to unify them. Secondly, there is already a Node in the engine, and without prefixing it, it is not possible to understand exactly what it is intended to do by looking at the name alone.

Reply:
Have prefixed them with "PoseGraph": https://github.com/cocos/cocos-engine/pull/14952/commits/9cc3c274925e8010d3f64fd5822523482d2cb14c